### PR TITLE
chore: remove lodash usage

### LINKED ISSUE #6489: Remove lodash usage
[Lodash](https://github.com/lodash/lodash) is only used once for it's [pick](https://github.com/search?q=repo%3AChainSafe%2Flodestar%20lodash&type=code) implementation. Remove it.

### PR DESCRIPTION
**Pull request by HiroyukiNaito**
**Motivation**

This is fix for the issue "Remove lodash usage #6489"

**Description**

Replace the lodash.pick for original pick function

Closes  #6489

**Steps to test or reproduce**

```sh
yarn build
yarn test:unit
```
-->

### CODE REVIEW

File: packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts [hunk 1]
```diff
@@ -274,3 +273,12 @@ function getMockDeposit({blockNumber, index}: {blockNumber: number; index: numbe
     depositData: {} as phase0.DepositData, // Not used
   };
 }
+
+function pick<T, K extends keyof T>(obj: T, keys: K[]): Pick<T, K> {
+  return keys.reduce((acc, key) => {
+      if (obj && Object.prototype.hasOwnProperty.call(obj, key)) {
```
Comment (jeluard) [inline packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: Any reason not to call `obj.hasOwnProperty` directly?

**Review by jeluard — CHANGES_REQUESTED** [review]
Looks like you need to run `lint:fix`, see output of GitHub checks.

File: packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts (hunk 1 above)
Comment (HiroyukiNaito) [reply packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: @jeluard Hey, sorry! My competent friend (ChatGPT) mistakenly changed my code.

**Comment by HiroyukiNaito** [comment]
> Looks like you need to run `lint:fix`, see output of GitHub checks.

Sure, did it and pushed. Please review it if anytime you have.

**Comment by jeluard** [comment]
Please run `yarn check-types` too, one of the checks is failing.

**Comment by HiroyukiNaito** [comment]
> Please run `yarn check-types` too, one of the checks is failing.

```TypeScript
@lodestar/beacon-node: $ tsc
@lodestar/beacon-node: test/unit/eth1/utils/eth1Data.test.ts:109:77 - error TS2345: Argument of type 'string[]' is not assignable to parameter of type '("depositRoot" | "depositCount" | keyof Eth1Block)[]'.
@lodestar/beacon-node:   Type 'string' is not assignable to type '"depositRoot" | "depositCount" | keyof Eth1Block'.
@lodestar/beacon-node: 109         const eth1DatasPartial = eth1Datas.map((eth1Data) => pick(eth1Data, Object.keys(expectedEth1Data[0])));
@lodestar/beacon-node:                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@lodestar/beacon-node: Found 1 error in test/unit/eth1/utils/eth1Data.test.ts:109
```

**Comment by HiroyukiNaito** [comment]
> Please run `yarn check-types` too, one of the checks is failing.

According to the Error. Changed  `string[]` assignable in args.

File: packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts [hunk 2]
```diff
@@ -274,3 +273,15 @@ function getMockDeposit({blockNumber, index}: {blockNumber: number; index: numbe
     depositData: {} as phase0.DepositData, // Not used
   };
 }
+
+function pick<T, K extends keyof T>(obj: T, keys: K[] | string[]): Pick<T, K> {
```
Comment (jeluard) [inline packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: Why do we need both input types? If we can't make it work with a generic `Pick`, I guess only `string[]` would do.

Comment (HiroyukiNaito) [reply packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: Yeah, a bit complicated.

Comment (HiroyukiNaito) [reply packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: @jaluard Corrected!

File: packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts [hunk 3]
```diff
@@ -274,3 +273,15 @@ function getMockDeposit({blockNumber, index}: {blockNumber: number; index: numbe
     depositData: {} as phase0.DepositData, // Not used
   };
 }
+
+function pick<T, K extends keyof T>(obj: T, keys: string[]): Pick<T, K> {
+  return (keys as K[]).reduce(
```
Comment (jeluard) [inline packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: Do we need this cast?
Also I am skeptical we need this `Pick` use here.

A different path would be to replca the `pick` usage altogether and replace it with something like:

```js
...
const eth1DatasPartial = eth1Datas.map(({blockNumber, depositCount}) => ({blockNumber, depositCount}));
...
```

Sorry for the back and forth!

Comment (HiroyukiNaito) [reply packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: I see. 
That's must be the answer for the simplest refactoring.

Comment (HiroyukiNaito) [reply packages/beacon-node/test/unit/eth1/utils/eth1Data.test.ts]: Your insights have really helped me improve my code. Thank you for your review.

**Review by nflaig — APPROVED** [review]
LGTM

**Review by jeluard — APPROVED** [review]

**Comment by jeluard** [comment]
Thanks @HiroyukiNaito for the effort!

**Comment by HiroyukiNaito** [comment]
Thanks for the dedicated support! I return the favor in the another issue.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.17.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.17.0) :tada:

**Pull request merged**